src/Microsoft.AspNetCore.Blazor/RenderTree/RenderTreeBuilder.cs

```diff
@@ -68,13 +68,19 @@ public void CloseElement()
         }
 
         /// <summary>
-        /// Appends a frame representing markup text content.
+        /// Appends a frame representing markup content.
         /// </summary>
         /// <param name="sequence">An integer that represents the position of the instruction in the source code.</param>
-        /// <param name="textContent">Content for the new markup text frame.</param>
-        public void AddMarkupContent(int sequence, string textContent) =>
-            // TODO stevesa implement me
-            throw null;
+        /// <param name="markupContent">Content for the new markup frame.</param>
+        public void AddMarkupContent(int sequence, string markupContent)
+        {
+            if (markupContent == null)
+            {
+                throw new ArgumentNullException(nameof(markupContent));
+            }
```

Comment:
Should we check non-null here? `new MarkupContent(string)` is happy to accept null. IMO we should permit both or neither

Reply:
Good point. It now just null-coalesces with `string.Empty` like we do for regular `string` values.